3rdparty/portmidi/pm_win/pmwinmm.c

```diff
@@ -729,8 +729,8 @@ static void FAR PASCAL winmm_in_callback(
             PmEvent event;
             if (midi->time_proc)
                 dwParam2 = (*midi->time_proc)(midi->time_info);
-            event.timestamp = dwParam2;
-            event.message = dwParam1;
+            event.timestamp = (PmTimestamp)dwParam2;
+            event.message = (PmMessage)dwParam1;
```

Comment:
Are `PmTimestamp` and `PmMessage` unsigned types? Just want to confirm because casting directly to an integer type differing in both size and signedness can behave unintuitively.

Reply:
Both are defined as int32_t in `3rdparty/portmidi/pm_common/portmidi.h`.

Reply:
Both params are always passed 32-bit values by Windows anyway, even when `DWORD_PTR` is 64 bits, so I don't think potentially casting back down to 32-bit types should affect anything.

Reply:
I guess `dwParam2` could become negative if you leave the device open for more than `INT_MAX` milliseconds, but I don't know if portmidi is even supposed to handle negative timestamps or not.

Reply:
2^31 milliseconds is a long time I guess.  Maybe we just hope for the best.